test/integration/scheduler/main_test.go

```diff
@@ -17,11 +17,123 @@ limitations under the License.
 package scheduler
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strings"
 	"testing"
 
+	"k8s.io/klog/v2/ktesting"
+	kubeschedulertesting "k8s.io/kubernetes/cmd/kube-scheduler/app/testing"
+	"k8s.io/kubernetes/test/integration"
 	"k8s.io/kubernetes/test/integration/framework"
 )
 
 func TestMain(m *testing.M) {
 	framework.EtcdMain(m.Run)
 }
+
+func TestHealthEndpoints(t *testing.T) {
+	server, apiserverConfig, _, err := integration.StartTestAPIServer(t)
+	if err != nil {
+		t.Fatalf("Failed to start kube-apiserver server: %v", err)
+	}
+	defer server.TearDownFn()
+
+	tests := []struct {
+		name             string
+		path             string
+		wantResponseCode int
+	}{
+		{
+			"/healthz",
+			"/healthz",
+			http.StatusOK,
+		},
+		{
+			"/livez",
+			"/livez",
+			http.StatusOK,
+		},
+		{
+			"/livez with ping check",
+			"/livez/ping",
+			http.StatusOK,
+		},
+		{
+			"/readyz",
+			"/readyz",
+			http.StatusOK,
+		},
+		{
+			"/readyz with informer-sync",
+			"/readyz/informer-sync",
+			http.StatusOK,
+		},
+		{
+			"/readyz with shutdown",
+			"/readyz/shutdown",
+			http.StatusOK,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt := tt
+			_, ctx := ktesting.NewTestContext(t)
+
+			result, err := kubeschedulertesting.StartTestServer(
+				ctx,
+				[]string{"--kubeconfig", apiserverConfig.Name(), "--leader-elect=false", "--authorization-always-allow-paths", tt.path})
+
+			if err != nil {
+				t.Fatalf("Failed to start kube-scheduler server: %v", err)
+			}
+			if result.TearDownFn != nil {
+				defer result.TearDownFn()
+			}
+
+			secureInfo := result.Config.SecureServing
+			secureOptions := result.Options.SecureServing
+			url := fmt.Sprintf("https://%s%s", secureInfo.Listener.Addr().String(), tt.path)
+			url = strings.Replace(url, "[::]", "127.0.0.1", -1) // switch to IPv4 because the self-signed cert does not support [::]
+
+			// read self-signed server cert disk
+			pool := x509.NewCertPool()
+			serverCertPath := path.Join(secureOptions.ServerCert.CertDirectory, secureOptions.ServerCert.PairName+".crt")
+			serverCert, err := os.ReadFile(serverCertPath)
+			if err != nil {
+				t.Fatalf("Failed to read component server cert %q: %v", serverCertPath, err)
+			}
+			pool.AppendCertsFromPEM(serverCert)
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: pool,
+				},
+			}
+
+			client := &http.Client{Transport: tr}
```

Comment:
Could L100-L119 be encapsulated into a separate function?

Reply:
It could, but I don't think the benefit of doing it now. Maybe we should do it when it's needed?

Reply:
As the test code gets a bit longer, I feel motivated to create a helper function for it now.